contracts/StakingVe.sol

```diff
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+/// @title A vesting style staking contract with extendable linear decay
+/// @author Auroter
+/// @notice Allows you to lock tokens in exchange for governance tokens
+/// @notice Locks can be extended or deposited into
+/// @notice Maximum deposit duration is two years (104 weeks)
+/// @dev Simply call stake(...) to create initial lock or extend one that already exists for the user
+
+// We have to maintain old versions of this to use legacy solidity (0.6.12)
+import "./libraries/SafeERC20.sol";
+import "./libraries/SafeMath.sol";
+import "./interfaces/IERC20.sol";
+
+//import "hardhat/console.sol";
+
+contract StakingVe {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // Info pertaining to staking contract
+    address private StakedToken; // An ERC20 Token to be staked (i.e. Axial)
+    address private Governance; // Who has administrative access
```

Comment:
What's the use of the governance as there are no administrative functions? Also let's use OZ's Ownable for access control instead.